### PR TITLE
[fix](dynamic-partition) fix wrong check of replication num

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

The following stmt will return error:
`errCode = 2, detailMessage = Failed to find 3 backends for policy: cluster|query|load|schedule|tags|medium: default_cluster|false|false|true|[{"location" : "default"}]|null`

```
CREATE TABLE if not exists `tbl1` (
`uuid` varchar(255),
`part` date,
)
DUPLICATE KEY(uuid)
PARTITION BY RANGE(part)()
DISTRIBUTED BY HASH(uuid) BUCKETS 32
PROPERTIES
(
"default_storage_medium" = "HDD",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "32",
"dynamic_partition.replication_num" = "1",
"dynamic_partition.create_history_partition"="true",
"dynamic_partition.start" = "-3"
);
```

This PR fix it.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.2 seconds
 load time: 568 seconds
 storage size: 17154655323 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028130831_clickbench_pr_35145.html

File: docs/zh-CN/docs/advanced/partition/dynamic-partition.md [hunk 1]
```diff
@@ -453,6 +453,87 @@ mysql> SHOW DYNAMIC PARTITION TABLES;
...
+    "dynamic_partition.start" = "-3",
+    "replication_num" = "3"
+    );
+    ```
+
+    这个示例中，没有创建任何初始分区（PARTITION BY 子句中的分区定义为空），并且设置了 `DISTRIBUTED BY HASH(k1) BUCKETS 3`, `"replication_num" = "3"`, `"dynamic_partition.replication_num" = "1"` 和 `"dynamic_partition.buckets" = "32"`。
+
+    我们将前两个参数成为表的默认参数，而后两个参数成为动态分区专用参数。
+
+    当系统自动创爱分区时，会使用分桶数 32 和 副本数 1 这两个配置（即动态分区专用参数）。而不是分桶数 3 和 副本数 3 这两个配置。
```
Comment (nextdreamblue) [inline docs/zh-CN/docs/advanced/partition/dynamic-partition.md:484]:
```suggestion
    当系统自动创建分区时，会使用分桶数 32 和 副本数 1 这两个配置（即动态分区专用参数）。而不是分桶数 3 和 副本数 3 这两个配置。
```

**Review by yiguolei — APPROVED** [review]
lgtm

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**